### PR TITLE
fix(java-ide): Make keymaps buffer specific

### LINKED ISSUE #42: use buffer only mappings when possible
We could use a simple function

```lua
on_attach = function(_, bufnr)
	local nmap = function(lhs, rhs, desc)
	  if desc then
	    desc = desc
	  end
	  
	  vim.keymap.set("n", lhs, rhs, { silent = true, desc = desc, buffer = bufnr, noremap = true })
	end
	
	local jtdls = require('jdtls')
	nmap("<leader>Jo", jdtls.organize_imports, "organize imports")
	nmap("<leader>Jv", jdtls.extract_variable, "extract variable")
	-- ...
end
```

### PR DESCRIPTION
**Pull request by dapc11**
This is desired in the scenario when you are working on multiple projects of different languages. With the old implementation you'll end up with dangling keymaps for language X when doing work for language Y.

### CODE REVIEW

**Comment by ChristianChiarulli** [comment]
pls resolve conflict, and if you could change `L` to `C` I would appreciate it

File: ftplugin/java.lua [hunk 1]
```diff
@@ -20,6 +20,54 @@ elseif vim.fn.has("unix") == 1 then
 else
 	print("Unsupported system")
 end
+local function set_keymaps(buffer)
```
Comment (kylo252) [inline ftplugin/java.lua]: please use this simplified function instead

```lua
on_attach = function(_, bufnr)
	local nmap = function(lhs, rhs, desc)
	  if desc then
	    desc = desc
	  end
	  
	  vim.keymap.set("n", lhs, rhs, { silent = true, desc = desc, buffer = bufnr, noremap = true })
	end
	
	local jtdls = require('jdtls')
	nmap("<leader>Jo", jdtls.organize_imports, "organize imports")
	nmap("<leader>Jv", jdtls.extract_variable, "extract variable")
	-- ...
end
```

Comment (dapc11) [reply ftplugin/java.lua]: @ChristianChiarulli what do you say? Perhaps this should be done for all branches? Currently the format is which-key format, which I think is fine. WDYT @kylo252?

Comment (kylo252) [reply ftplugin/java.lua]: We'd really appreciate it if you could manage to fix all the branches, that would be really helpful!!

Comment (dapc11) [reply ftplugin/java.lua]: I'll see what I can do. :)

**Comment by dapc11** [comment]
> pls resolve conflict, and if you could change `L` to `C` I would appreciate it

Done.

**Pull request merged**